accepted/2024/end-of-life-packages.md

```diff
@@ -0,0 +1,110 @@
+| EOL         | No longer maintained or supported       | No                | Migrate to alternatives                | High     |
+| Vulnerable  | Contains known security vulnerabilities | Possibly          | Update or find secure alternatives     | High     |
+| Deprecated  | Not recommended for use                 | Possibly          | Transition to newer packages           | Moderate |
+
+## Motivation 
+
+<!-- Why are we doing this? What pain points does this solve? What is the expected outcome? -->
+
+There are two primary motivations in doing this work. The first being a direct answer to the "am I supported?" question. This is especially prevalent in the .NET ecosystem due to many first-party package providers have robust support policies that require a developer to understand with a decoder ring and too much time on their hands. It should be explictly simple to know you're supported in your current solution so long as your packages do not report EOL.
+
+The second primary motivation is security. Being on unsupported packages that are EOL is a high risk as there will be no security updates and may render your application vulnerable. A typical lifecycle of an end of life package may accrue known vulnerabilities over the years as they are discovered which may add additional risk for developer teams to sincerely consider prompt remediation.
+
+The expected outcome would be developers having a clear package status for knowing if a package in their package graph is "supported" or not through its status being EOL. EOL packages tend to be very difficult for engineering teams to build in the first place to provide security updates and typically aren't serviced do to the difficulty.
+
+## Explanation
+
+### Functional explanation
+
+<!-- Explain the proposal as if it were already implemented and you're teaching it to another person. -->
+<!-- Introduce new concepts, functional designs with real life examples, and low-fidelity mockups or  pseudocode to show how this proposal would look. -->
+This functionality will follow the footsteps of the existing deprecation and vulnerability functionality. It will be closest to the deprecation experience today, but will include a way for a package owner to mark a package and its many versions at anytime "EOL". There will be no reason nor custom message to provide to users. However, there will be an suggested alternate package to use to encourage a direct action from the developer.
+
+When a package is marked EOL, a few things will happen:
+
+- The package at restore time will appear with a new warning letting any consumers know the package is EOL.
+- The package on NuGet.org and throughout browsing experiences will show a label or affordance mentioning its EOL status.
+- The package on NuGet.org API can be queried for the EOL package status for any tooling to build upon this.
+- The dotnet CLI will have a list command to show all EOL packages.
```

Comment:
How can we accommodate customers who want to suppress the display of EOL packages from the configured package feeds? It could be a nice way to prevent installation of them. Imagine a *nuget.config* file placed at the root of the project repository with a setting that enables this behavior. Any dev working on that project won't see EOL packages when browsing configured package feeds.

Reply:
Ideally they can just disable auditing type flags within their project similar to what vulnerability provides today

https://devblogs.microsoft.com/nuget/nugetaudit-2-0-elevating-security-and-trust-in-package-management/